support-frontend/assets/helpers/subscriptionsForms/submit.js

```diff
@@ -80,42 +76,40 @@ function getAddresses(state: AnyCheckoutState) {
   };
 }
 
-const getOptions = (
-  fulfilmentOptions: FulfilmentOptions,
-  productOptions: ProductOptions,
-): {fulfilmentOptions: FulfilmentOptions, productOptions: ProductOptions} =>
-  ({
-    ...(fulfilmentOptions !== NoFulfilmentOptions ? { fulfilmentOptions } : {}),
-    ...(productOptions !== NoProductOptions ? { productOptions } : {}),
-  });
-
 const getProduct =
-  (state: AnyCheckoutState, currencyId?: Option<IsoCurrency>): DigitalSubscription | PaperSubscription => {
+  (state: AnyCheckoutState, currencyId?: Option<IsoCurrency>): SubscriptionProductFields => {
     const {
       billingPeriod,
       fulfilmentOption,
       productOption,
       orderIsAGift,
       product,
-      giftMessage,
-      giftDeliveryDate,
     } = state.page.checkout;
 
     const readerType = orderIsAGift ? Gift : Direct;
     if (product === DigitalPack) {
       return {
+        productType: 'DigitalPack',
```

Comment:
```suggestion
        productType: DigitalPack,
```

Reply:
I don't think flow  can really link the if statement with the contents of `product` in this way.  I think since this is an external api it is fine to be explicit.  Unless there's a neater way I don't know about?
![image](https://user-images.githubusercontent.com/7304387/101049533-85fafc80-357b-11eb-8423-9e39d2f0e075.png)

Reply:
That's odd I would have thought that it would realise we'd checked that. Ah well...

Reply:
We should still update these to use the const in subscriptions.js rather than using strings, so:
` productType: DigitalPack,`

Reply:
onwards and upwards to typescript !